File: src/CorporateMessengerWithAgile/Application/Dto/TaskItemInSprintDto.cs
using Domain.Entity;

namespace Application.Dto
{
    public class TaskItemInSprintDto : BaseDto
    {
        public Guid TaskItemId { get; set; }
        public Guid SprintId { get; set; }
        public TaskItemStatus TaskStatus { get; set; }
        public string Description { get; set; } = null!;
    }
}

File: src/CorporateMessengerWithAgile/ReactApp/reactapp.client/src/models/entity/TaskItemInSprintDto.ts
// src/models/entity/TaskItemInSprintDto.ts
import { z } from 'zod';
import { BaseDtoSchema } from './BaseDto';
import { GuidSchema } from '../Guid';

export const TaskItemInSprintDtoSchema = BaseDtoSchema.extend({
    taskItemId: GuidSchema,
    sprintId: GuidSchema,
    taskStatus: z.string().min(1),
    description: z.string(),
});

export type TaskItemInSprintDto = z.infer<typeof TaskItemInSprintDtoSchema>;
